File: app/src/main/java/com/solarapp/filtersearch/data/NewsDao.java
package com.solarapp.filtersearch.data;

import androidx.lifecycle.MutableLiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface NewsDao {
    @Insert(onConflict = REPLACE)
    void insertNews(News news);

    @Update(onConflict = REPLACE)
    void updateNews(News news);

    @Delete
    void deleteNews(News news);

    @Query("SELECT * FROM News")
    public MutableLiveData<List<News>> findAllNewsSync();
}

File: app/src/main/java/com/team/veza/googlenew/database/NewsDao.kt
package com.team.veza.googlenew.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.team.veza.googlenew.model.News

@Dao
interface NewsDao {
    @Query("Select * from tbl_news")
    fun get():LiveData<List<News>>
    @Query("Select * from tbl_news where title like :key")
    fun get(key:String):LiveData<List<News>>
    @Query("Select * from tbl_news where isFavorite=1")
    fun getFavorite():LiveData<List<News>>
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(news:News):Long
    @Delete
    fun delete(news:News)
    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun update(news:News)
}